yt/data_objects/construction_data_containers.py

```diff
@@ -684,6 +687,46 @@ def __init__(
             + self.ds.domain_offset
         )
         self._setup_data_source()
+
+        # determine if we need to extrapolate data for non-periodic domains
+        for i in range(3):
+            if not self.ds.periodicity[i] and (
+                self.left_edge[i] < self.ds.domain_left_edge[i]
+                or self.right_edge[i] > self.ds.domain_right_edge[i]
+            ):
+                self.fix_nonperiodic[i] = True
+            if self.fix_nonperiodic[i]:
```

Comment:
Seems redundant, or am I missing something ?
```suggestion
```
Also, I find this block hard to read, it would help to invert some early condition checks to save indentation levels if we can

Reply:
Agreed, I will fix this!